static/src/javascripts/projects/common/modules/crosswords/main.js

```diff
@@ -597,7 +597,13 @@ define([
         },
 
         hasSolutions: function () {
```

Comment:
I'm not sure but I think you might need to wrap the new logic in an if('solution' in this.props.data.entries[0]). It's unlikely but I think it's possible that a crossword could have a dateSolutionAvailable in the past or solutionAvailable set to true but there to be no actual solution available in the response from capi.